docs/aws/audit/kubernetesmonitoring/rules/efs_access_point_enforce_root_directory.mdx

```diff
@@ -23,6 +23,219 @@ CBP,RBI_MD_ITF
+    print("All EFS Access Points enforce the root directory.")
+else:
+    print("Some EFS Access Points do not enforce the root directory.")
+
+# Enforce root directory in Kubernetes PV
+enforce_root_directory_in_pv(k8s_client, pv_name)
+```
+
+### Summary
+1. **Install Required Libraries**: Ensure `boto3` and `kubernetes` libraries are installed.
+2. **Set Up AWS and Kubernetes Clients**: Initialize the clients for AWS and Kubernetes.
+3. **Check EFS Access Points Configuration**: Create a function to check if EFS Access Points enforce the root directory.
+4. **Enforce Root Directory in Kubernetes Persistent Volume**: Create a function to ensure the PV in Kubernetes uses the root directory of the EFS.
+
+By following these steps, you can prevent the misconfiguration of EFS Access Points not enforcing the root directory in Kubernetes using Python scripts.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Log in to the AWS Management Console and navigate to the EFS service.
+
+2. In the EFS dashboard, select the file system that you want to check.
+
+3. In the file system details page, click on the "Access points" tab.
+
+4. For each access point, check the "Root directory creation permissions" section. If the "Owner" field is not set to 0 (root), then the EFS access point is not enforcing root directory.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Import the necessary libraries: You will need the boto3 library, which allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, etc. You can install it using pip:
+
+```python
+pip install boto3
+```
+
+2. Establish a session: You need to establish a session using your AWS credentials. 
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    aws_session_token='SESSION_TOKEN',
+)
+```
+
+3. Connect to the EFS service: Once the session is established, you can connect to the EFS service and retrieve all the file systems.
+
+```python
+efs = session.client('efs')
+file_systems = efs.describe_file_systems()
+```
+
+4. Check the root directory enforcement: For each file system, you need to check if the root directory is enforced. You can do this by checking the 'RootDirectory' attribute of each access point.
+
+```python
+for fs in file_systems['FileSystems']:
+    access_points = efs.describe_access_points(FileSystemId=fs['FileSystemId'])
+    for ap in access_points['AccessPoints']:
+        if 'RootDirectory' not in ap or 'Path' not in ap['RootDirectory'] or ap['RootDirectory']['Path'] != '/':
+            print(f"Access point {ap['AccessPointId']} in file system {fs['FileSystemId']} does not enforce root directory.")
+```
+
+This script will print out the IDs of all access points that do not enforce the root directory.
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>